packages/test/test-end-to-end-tests/src/test/gc/gcDataVirtualization.spec.ts

```diff
@@ -0,0 +1,219 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import {
+	createTestConfigProvider,
+	createSummarizer,
+	ITestContainerConfig,
+	ITestObjectProvider,
+	summarizeNow,
+	waitForContainerConnection,
+} from "@fluidframework/test-utils";
+import {
+	describeCompat,
+	ITestDataObject,
+	TestDataObjectType,
+} from "@fluid-private/test-version-utils";
+import {
+	ISummaryTree,
+	SummaryType,
+	type ISnapshotTree,
+} from "@fluidframework/protocol-definitions";
+import { IContainer, LoaderHeader } from "@fluidframework/container-definitions";
+import type { ISnapshot } from "@fluidframework/driver-definitions";
+import { getGCStateFromSummary } from "./gcTestSummaryUtils.js";
+
+const interceptResult = <T>(
+	parent: any,
+	fn: (...args: any[]) => Promise<T>,
+	intercept: (result: T) => void,
+) => {
+	const interceptFn = async (...args: any[]) => {
+		const val = await fn.apply(parent, args);
+		intercept(val);
+		return val as T;
+	};
+	parent[fn.name] = interceptFn;
+	interceptFn.bind(parent);
+	return fn;
+};
```

Comment:
By the 'Rule of three', this should probably go into some sort of 'common' file. Not right now, I'd say this is orthogonal to the PR.

Reply:
Agreed, I just haven't gotten around to it. Not sure which common file is recommended, I'll figure it out.